mlir/test/Dialect/LLVMIR/genx.mlir

```diff
@@ -119,3 +119,10 @@ func.func @genx.matrix.load(%ptr : !llvm.ptr<vector<4xi32>>, %stride : index) {
   %0 = genx.matrix.load <Subgroup> <RowMajor> %ptr, %stride {memory_access = #genx.memory_access<Volatile>} : (!llvm.ptr<vector<4xi32>>, index) -> !genx.jointmatrix<8x16xi32, RowMajor>
   llvm.return  
 }
+
+func.func @genx.matrix.mad(%a : !genx.jointmatrix<8x32xi8, RowMajor>, %b : !genx.jointmatrix<32x8xi8, ColumnMajor>, %c : !genx.jointmatrix<8x8xi32, RowMajor>) {
+  // CHECK-LABEL: genx.matrix.mad
+  // CHECK: %0 = genx.matrix.mad <Subgroup> %arg0, %arg1, %arg2 : !genx.jointmatrix<8x32xi8, RowMajor>, !genx.jointmatrix<32x8xi8, ColumnMajor> -> !genx.jointmatrix<8x8xi32, RowMajor>
+  %0 = genx.matrix.mad <Subgroup> %a, %b, %c : !genx.jointmatrix<8x32xi8, RowMajor>, !genx.jointmatrix<32x8xi8, ColumnMajor> -> !genx.jointmatrix<8x8xi32, RowMajor>
```

Comment:
is that allowed to add matrix of `i8` with matrix of `i32`?

Reply:
Yes the combination is valid (https://github.com/intel/llvm/blob/54cd00d559a89931e4bcc5cf0dbf5938e2767c52/sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc#intel-xmx-supported-combinations)